chapter-11/end/Levels/LevelReader.cs

```diff
@@ -26,8 +26,12 @@ public List<List<BaseGameStateEvent>> LoadLevel(int nb)
             var fileName = $"{assemblyName}.Levels.LevelData.Level{nb}.txt";
 
             var stream = assembly.GetManifestResourceStream(fileName);
-            var reader = new StreamReader(stream);
-            var levelString = reader.ReadToEnd();
+
+            string levelString;
+            using (var reader = new StreamReader(stream))
+            {
+                levelString = reader.ReadToEnd();
+            }
 
             var rows = levelString.Split(Environment.NewLine.ToCharArray(), StringSplitOptions.RemoveEmptyEntries);
```

Comment:
Maybe a conditional null check here in case levelString is still null? Or are we handling the null a layer above?

Reply:
I'm having project fatigue and getting sloppy. Good call. Fixed.